db/users.go

```diff
@@ -0,0 +1,85 @@
+package db
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/lbryio/lbrytv/config"
+	"github.com/lbryio/lbrytv/lbrynet"
+	"github.com/lbryio/lbrytv/models"
+	"github.com/lbryio/lbrytv/monitor"
+
+	"github.com/lbryio/lbry.go/extras/lbryinc"
+	"github.com/volatiletech/sqlboiler/boil"
+)
+
+// TokenHeader is the name of HTTP header which is supplied by client and should contain internal-api auth_token
+const TokenHeader string = "X-Lbry-Auth-Token"
+
+// GetUserByToken retrieves user by internal-api auth_token
+// TODO: Refactor out into logical pieces
+// TODO: Implement different error types for better error messages
+func GetUserByToken(token string) (*models.User, error) {
+	var u *models.User
+	ctx := context.Background()
+
+	u, err := models.Users(models.UserWhere.AuthToken.EQ(token)).OneG(ctx)
+	if err != nil {
+		Conn.Logger.LogF(monitor.F{"token": token}).Info("token supplied but isn't present in the database, trying internal-apis")
```

Comment:
hmmm, should we really be logging user tokens? This seems like a security issue. I would just log their user id or something.

Reply:
btw this is in multiple areas, just referencing it once though.